src/app/code/community/Omise/Gateway/Model/Payment/Offsitecitipoints.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+
+class Omise_Gateway_Model_Payment_Offsitecitipoints extends Omise_Gateway_Model_Payment_Base_Payment {
+
+    /**
+     * @var string
+     */
+    protected $_code = 'omise_offsite_citipoints';
+
+    /**
+     * @var string
+     */
+    protected $_formBlockType = 'omise_gateway/form_offsitecitipoints';
+
+    /**
+     * @var string
+     */
+    protected $_infoBlockType = 'payment/info';
+
+    /**
+     * @var string
+     */
+    protected $_callbackUrl = 'omise/callback_validateoffsitecitipoints';
+
+    /**
+     * @var array
+     */
+    protected $_currencies = array('THB');
+
+    /**
+     * Payment Method features
+     *
+     * @var bool
+     */
+    protected $_isGateway          = true;
+    protected $_canReviewPayment   = true;
+    protected $_isInitializeNeeded = true;
+
+    /**
+     * @param  Varien_Object $payment
+     * @param  float         $amount
+     *
+     * @return Omise_Gateway_Model_Api_Charge
+     */
+    public function process(Varien_Object $payment, $amount)
```

Comment:
Looks like there might be some scope for more refactoring here since Citi and Alipay seem to have all but identical `process` methods. Maybe an intermediary sub class from the base one (Omise_Gateway_Model_Payment_SimpleOffsite_Payment?)

Thoughts @jacstn ?